tests/run/int.rs

```diff
@@ -3,151 +3,343 @@
 // Run-time:
 //   status: 0
 
-#![feature(arbitrary_self_types, auto_traits, core_intrinsics, lang_items, start, intrinsics)]
+#![feature(bench_black_box, core_intrinsics, start)]
 
 #![no_std]
 
-mod intrinsics {
-    extern "rust-intrinsic" {
-        pub fn abort() -> !;
-    }
+#[panic_handler]
+fn panic_handler(_: &core::panic::PanicInfo) -> ! {
+    core::intrinsics::abort();
 }
 
 /*
- * Core
+ * Code
  */
 
-mod libc {
-    #[link(name = "c")]
-    extern "C" {
-        pub fn puts(s: *const u8) -> i32;
+#[start]
+fn main(_argc: isize, _argv: *const *const u8) -> isize {
+    let one: isize = core::hint::black_box(1);
+
+    macro_rules! check {
+        ($ty:ty, $expr:expr) => {
+            {
+                const EXPECTED: $ty = {
+                    #[allow(non_upper_case_globals)]
+                    #[allow(dead_code)]
+                    const one: isize = 1;
```

Comment:
Since we switched to `black_box`, `one` is not needed anymore. If you don't want to remove it and adjust all numbers in this PR, though, it's ok, I can do it later.

Reply:
Did you mean something like this? https://github.com/rust-lang/rustc_codegen_gcc/commit/a7a09d556ab222f3b24d76defe1d2b2f99aa2c6f

Reply:
Yes, thanks!